src/main/java/patternverifiers/SingletonVerifier.java

```diff
@@ -1,17 +1,312 @@
+        boolean instanceMethod = true;
+        List<MethodDeclaration> methods = new ArrayList<>();
+        compUnit.findAll(MethodDeclaration.class).forEach(
+            methodDeclaration -> {  //Make a list of all
+                // methods in the class.
+                methods.add(methodDeclaration);
+            });
+        for (MethodDeclaration declaration : methods) { // For each method
+            if (declaration.getTypeAsString().equals(
+                compUnit.getPrimaryTypeName().get())) {   // If the method
+                // returns an instance of the Singleton
+                if (declaration.isStatic()) {  // If the method is static
+                    if (declaration.isPrivate()) {  // If the method is private
+                        compUnit.findAll(ConstructorDeclaration.class).forEach(
+                            constructor -> constDeclList.add(constructor));
+                        instanceMethod &= isMethodCalledFromPublic(methods, declaration);   //
+                        // Check if the method is called from a public method
+                        // (isMethodCalledFromPublic calls itself recursively to find nested
+                        // method calls)
+                        // return findConstructorCall(declaration, constructorDeclaration);
+                    } else {
+                        instanceMethod &= true; // AND with true
+                    }
+                } else {
+                    instanceMethod &= false;    // AND with false
+                }
+            }
+        }
+        return instanceMethod;  // Return result
+    }
+
+    /**
+     * Method to look for a constructor call, might be useful to differentiate between different
+     * types of singletons in the future, currently unused.
+     *
+     * @param declaration A method to look inside to see if there is a ConstructorCall
+     * @param constructor A ConstructorDeclaration for the class
+     *
+     * @return If a method is calling a specific constructor
+     */
+    public boolean findConstructorCall(
+        MethodDeclaration declaration, ConstructorDeclaration constructor) {
+        List<ObjectCreationExpr> calledMethods = new ArrayList<>();
+        declaration.findAll(ObjectCreationExpr.class).forEach(methodDeclaration -> {    // Find all
+            // ObjectCreations in the method
+            if (methodDeclaration.getTypeAsString().equals(
+                constructor.getNameAsString())) {    // If the object created is an instance
+                // of the singleton
+                calledMethods.add(methodDeclaration);   // Add the ObjectCreationExpr to
+                // the list of ObjectCreationExpr
+            }
+        });
+        return !calledMethods.isEmpty(); // If the list of ObjectCreationExpr is empty -> return
+        // true
+    }
+
+
+    // (not 100% sure it doesn't end up in an infinite loop if it never finds a public call,
```

Comment:
This should be investigated and not part of the javadoc. Documenting possible bugs feels weird.

Reply:
Resolved in next commit